CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiDevice.cpp

```diff
@@ -0,0 +1,239 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include "windows_devices.h"
+
+typedef Library_windows_devices_native_Windows_Devices_Spi_SpiConnectionSettings SpiConnectionSettings;
+
+// NOTE: Must match enum in SpiEnums.cs
+enum SpiMode
+{
+        SpiMode_Mode0 = 0,
+        SpiMode_Mode1,
+        SpiMode_Mode2,
+        SpiMode_Mode3,
+};
+
+HRESULT Library_windows_devices_native_Windows_Devices_Spi_SpiDevice::InitNative___VOID( CLR_RT_StackFrame& stack )
+{
+    TINYCLR_HEADER(); hr = S_OK;
+
+    GPIO_PIN chipSelectPin = GPIO_PIN_NONE;
+    GPIO_PIN mskPin = GPIO_PIN_NONE;
+    GPIO_PIN misoPin = GPIO_PIN_NONE;
+    GPIO_PIN mosiPin = GPIO_PIN_NONE;
+    bool chipSelectReserved = false;
+    bool mskPinReserved = false;
+    bool misoPinReserved = false;
+    bool mosiPinReserved = false;
+    CLR_RT_HeapBlock* settings;
+
+    CLR_RT_HeapBlock* pThis = stack.This(); FAULT_ON_NULL(pThis);
+
+    if (pThis[ FIELD__m_disposed ].NumericByRef().u1)
+    {
+        TINYCLR_SET_AND_LEAVE(CLR_E_OBJECT_DISPOSED);
+    }
+
+    settings = pThis[ FIELD__m_settings ].Dereference(); FAULT_ON_NULL(settings);
+    chipSelectPin = settings[ SpiConnectionSettings::FIELD__m_chipSelectionLine ].NumericByRef().u4;
+
+    // Get the pin IDs for this device's chip select line. Then reserve all pins which have assigned IDs.
+    CPU_SPI_GetPins( pThis[ FIELD__m_spiBus ].NumericByRef().u4, mskPin, misoPin, mosiPin );
+
+    if (chipSelectPin != GPIO_PIN_NONE)

```

Comment:
We have to fight the Visual Studio defaults on this one, and the code base is inconsistent.  Pushing back unless you feel very strongly about it.